libs/providers/flagd/src/lib/service/in-process/grpc/grpc-fetch.ts

```diff
@@ -0,0 +1,99 @@
+import {DataFetch} from "../data-fetch";
+import {Config} from "../../../configuration";
+import {FlagSyncServiceClient, SyncFlagsRequest, SyncFlagsResponse} from "../../../../proto/ts/sync/v1/sync_service";
+import {ClientReadableStream, credentials} from "@grpc/grpc-js";
+
+export const initBackOffMs = 2 * 1000;
+const maxStartupDeadlineMs = 500;
+const maxBackOffMs = 120 * 1000;
+
+/**
+ * Implements the gRPC sync contract to fetch flag data.
+ */
+export class GrpcFetch implements DataFetch {
+  private _connecting = false;
+  private _nextBackoff = initBackOffMs;
+  private _syncClient: FlagSyncServiceClient;
+  private _syncStream: ClientReadableStream<SyncFlagsResponse> | undefined;
+  private readonly _request: SyncFlagsRequest;
+
+  constructor(config: Config, syncServiceClient ?: FlagSyncServiceClient) {
+    const {host, port, tls, socketPath} = config;
+
+    this._syncClient = syncServiceClient ? syncServiceClient : new FlagSyncServiceClient(
+      socketPath ? `unix://${socketPath}` : `${host}:${port}`,
+      tls ? credentials.createSsl() : credentials.createInsecure());
+
+    this._request = {providerId: "", selector: config.selector ? config.selector : ''};
```

Comment:
Is it not possible to set a `providerId`? Can this be omitted?

Reply:
You should also be able to extract the selector from the config on like 21 and set a default if it's missing.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#syntax

Reply:
unfortunately `providerId ` is not set as optional in the sync contract. And available constructors anyway set this to empty string. I couldn't find any other alternative here. 

Other change on extraction is done/